bdb/file.c

```diff
@@ -4719,13 +4722,10 @@ static int bdb_downgrade_int(bdb_state_type *bdb_state, int noelect,
         *downgraded = 0;
 
     retries = 0;
```

Comment:
This addresses another "lxc-locked-cluster" error at startup.  A node would win an election under egen x, but get downgraded by a subsequent election before it managed to call upgrade.  The upgrade_allowed flag is set near the end of bdb_open_int.  The retry/poll loop allowed this code (which is holding the bdb-writelock) to prevent the thread which is attempting to call 'upgrade' from getting the bdb writelock.  We can allow it to upgrade even though it is no longer master because none of it's log-writes will be acknowledged by the rest of the cluster, and it's initial 'NEWMASTER' message (called by virtue of the upgrade codepath's __rep_start) will be immediately met with a DUPMASTER from the clusters actual master.

Reply:
We don't entirely understand this .. may not need the upgrade_allowed flag at all.